tests/unittests/unittests.cpp

```diff
@@ -4264,6 +4269,137 @@ class ConcurrentQueueTests : public TestClass<ConcurrentQueueTests>
 		
 		return true;
 	}
+
+	bool acquire_and_signal()
+	{
+		typedef ::moodycamel::LightweightSemaphore semaphore;
+
+		// Test resource acquisition from one other thread.
+		{
+			semaphore s;
+			s.signal(); // Single resource available
+						
+			const auto fnTestSingleAcquire = 
+				[](::moodycamel::LightweightSemaphore &s) -> bool
+				{
+					for (std::size_t k = 0; k < 100; ++k)
```

Comment:
100 iterations is likely not enough to force concurrency (the overhead of starting a new thread will eclipse the time the thread takes to execute).